src/Toml/Parser/TOML.hs

```diff
@@ -56,54 +52,73 @@ tableNameP = between (text "[") (text "]") keyP
 tableArrayNameP :: Parser Key
 tableArrayNameP = between (text "[[") (text "]]") keyP
 
--- | Parser for lines starting with 'key =', either values or inline tables.
-hasKeyP :: Parser (Key, Either AnyValue TOML)
-hasKeyP = (,) <$> keyP <* text "=" <*> eitherP anyValueP inlineTableP
+-- Helper functions for building TOML
+tomlKV :: Key -> AnyValue -> TOML
+tomlKV k v = mempty { tomlPairs = HashMap.singleton k v}
+
+tomlT :: Key -> TOML -> TOML
+tomlT k t = mempty {tomlTables = single k t}
+
+tomlA :: Key -> NonEmpty TOML -> TOML
+tomlA k a = mempty {tomlTableArrays = HashMap.singleton k a}
+
+-- | Parser for lines starting with 'key =', either values, inline tables or
+-- inline arrays of tables.
+hasKeyP :: Maybe Key -> Parser TOML
+hasKeyP key = do
+    k <- keyP <* text "="
+    try (tableArray k) <|> try (table k) <|> (tomlKV k <$> anyValueP)
```

Comment:
Will it work without these two `try`s here? THere is a `try` inside `childKeyP`

Reply:
Mmh, maybe the one around `table` isn't needed because only an inline table starts with `{`, the one around `tableArray` is needed. Also note that here `childKeyP` will never consume any input because the parser is `pure k`, so the `try` inside is useless.